creator-node/src/RehydrateIpfsQueue.js

```diff
@@ -3,14 +3,62 @@ const { rehydrateIpfsFromFsIfNecessary, rehydrateIpfsDirFromFsIfNecessary } = re
 const { logger: genericLogger } = require('./logging')
 const config = require('./config')
 const enableRehydrate = config.get('enableRehydrate')
+const redisClient = require('./redis')
 
 const PROCESS_NAMES = Object.freeze({
   rehydrate_dir: 'rehydrate_dir',
   rehydrate_file: 'rehydrate_file'
 })
 
+// minimum threshold per day to rehydrate
+const MIN_REHYDRATE_THRESHOLD = config.get('minRehydrateThreshold') || 10
 const MAX_COUNT = 10000
 
+class RehydrateRedisCounter {
+  constructor (redis) {
+    this.redis = redis
+  }
+
+  /**
+   * A redis key that stores hash of CID -> count and is incremented when we get a request for the CID
+   * A new key is created for each day and is automatically deleted 24 hours after the final write
+   * @param {String} taskType type of rehydrate task eg 'addRehydrateIpfsFromFsIfNecessaryTask' or 'addRehydrateIpfsDirFromFsIfNecessaryTask'
+   * @returns {String} redis key
+   */
+  static constructCounterRedisKey (taskType) {
+    return `${taskType}:::counter:::${new Date().toISOString().split('T')[0]}`
+  }
+
+  // keep count in redis for the number of times rehydrate is requested for each CID
+  async incrementCount (taskType, CID) {
+    const count = await this.redis.hincrby(RehydrateRedisCounter.constructCounterRedisKey(taskType), CID, 1)
+    await this.redis.expire(RehydrateRedisCounter.constructCounterRedisKey(taskType), 60 * 60 * 24) // expire one day after final write
+    return count
+  }
+
+  /**
+   * A redis key that stores a set of CID's that have been rehydrated that day
+   * A new key is created for each day and is automatically deleted 24 hours after the final write
+   * @param {String} taskType type of rehydrate task eg 'addRehydrateIpfsFromFsIfNecessaryTask' or 'addRehydrateIpfsDirFromFsIfNecessaryTask'
+   * @returns {String} redis key
+   */
+  static constructRehydrateCompleteRedisKey (taskType) {
+    return `${taskType}:::rehydrateComplete:::${new Date().toISOString().split('T')[0]}`
+  }
+  // add CID to set of CIDs that have been rehydrated today
+  async addToRehydratedSet (taskType, CID) {
+    await this.redis.sadd(RehydrateRedisCounter.constructRehydrateCompleteRedisKey(taskType), CID)
```

Comment:
same here -- move key creation into a const in fn scope

Reply:
done